bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -611,14 +592,21 @@ impl Room {
         Ok(self.inner.set_unread_flag(new_value).await?)
     }
 
-    /// Mark a room as read, by attaching a read receipt on the latest event.
+    /// Mark a room as read, by attaching a read receipt on the latest timeline
+    /// item.
     ///
     /// Note: this does NOT unset the unread flag; it's the caller's
     /// responsibility to do so, if needs be.
     pub async fn mark_as_read(&self, receipt_type: ReceiptType) -> Result<(), ClientError> {
-        let timeline = self.timeline().await?;
+        // Create a throwaway timeline for this.
+        let timeline = SdkTimeline::builder(&self.inner)
+            .track_read_marker_and_receipts()
+            .build()
+            .await
```

Comment:
Is there a risk that the creation of the `Timeline` and the installation of all its tasks, take a lot of time? Maybe we should add a big warning in the doc?

Reply:
We know it can be slow, if there are many events, especially because of read receipts, so yeah I'll add a comment :+1: